### ISSUE TITLE
False negative: popping a ReadOnly key from a TypedDict

### ISSUE THREAD

**Issue by JelleZijlstra** [python/mypy#19130]
Mypy reports no error for this program:

```python
from typing import ReadOnly, TypedDict, NotRequired

class TD(TypedDict):
    x: ReadOnly[NotRequired[int]]
    
def f(td: TD):
    td.pop("x")
```

https://mypy-play.net/?mypy=latest&python=3.13&gist=9dfad53c9dc2e22c63548a553afdb67e

But deleting a ReadOnly key should not be allowed. The [spec](https://typing.python.org/en/latest/spec/typeddict.html#typing-readonly-type-qualifier) says:

> The typing.ReadOnly [type qualifier](https://typing.python.org/en/latest/spec/glossary.html#term-type-qualifier) is used to indicate that an item declared in a TypedDict definition may not be mutated (added, modified, or removed)
